lib/internal/modules/esm/load.js

```diff
@@ -129,18 +129,22 @@ async function defaultLoad(url, context = { __proto__: null }) {
 
   if (urlInstance.protocol === 'node:') {
     source = null;
-  } else if (source == null) {
-    ({ responseURL, source } = await getSource(urlInstance, context));
-    context.source = source;
-  }
+    format ??= 'builtin';
+  } else {
+    let contextToPass = context;
+    if (source == null) {
+      ({ responseURL, source } = await getSource(urlInstance, context));
+      contextToPass = { __proto__: context, source };
```

Comment:
Why `__proto__: context`?

```suggestion
      contextToPass = { __proto__: null, source };
```

Reply:
To inherit the properties from the hook-provided `context`. `null` would not make sense IMO